crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -121,6 +121,19 @@ def checks
             upgrade_status.end_step
           end
         end
+      rescue ::Crowbar::Error::StartStepRunningError,
+             ::Crowbar::Error::StartStepOrderError,
+             ::Crowbar::Error::SaveUpgradeStatusError => e
+        raise ::Crowbar::Error::UpgradeError.new(e.message)
+      rescue StandardError => e
+        ::Crowbar::UpgradeStatus.new.end_step(
```

Comment:
IMO this should be done in all cases. Otherwise if you have something running fine and you try to start it again it will set the status to failed, but the step is still running. In case of 'running' status the response should say 'failed' but the status should still say 'running'. For async steps the trigger endpoint should not succeed in such cases.